src/Microsoft.ApiDesignGuidelines.Analyzers/Microsoft.ApiDesignGuidelines.Analyzers.md

```diff
@@ -211,6 +339,44 @@ Severity: Warning
 
 Help: [https://msdn.microsoft.com/en-us/library/ms182253.aspx](https://msdn.microsoft.com/en-us/library/ms182253.aspx)
 
+### CA1724: Type names should not match namespaces ###
+
+Type names should not match the names of namespaces that are defined in the .NET Framework class library. Violating this rule can reduce the usability of the library.
+
```

Comment:
Another point is that (as @srivastn pointed out) ".NET Framework" isn't as well-defined a term as it used to be. There is a certain hard-coded set of common Framework namespaces that the analyzer looks for. I don't know if you can easily express that in the description (you certainly don't want to enumerate the whole set in the description).